runtime/wasm/src/host.rs

```diff
@@ -684,103 +689,177 @@ impl RuntimeHostTrait for RuntimeHost {
         log: Arc<Log>,
         state: BlockState,
     ) -> Box<Future<Item = BlockState, Error = Error> + Send> {
-        // Identify event handler for this log
-        let event_handler = match self.handler_for_log(&log) {
-            Ok(handler) => handler,
-            Err(e) => return Box::new(future::err(e)),
-        };
+        let logger = logger.clone();
+        let mapping_request_sender = self.mapping_request_sender.clone();
 
-        // Identify the event ABI in the contract
-        let event_abi = match util::ethereum::contract_event_with_signature(
-            &self.data_source_contract_abi.contract,
-            event_handler.event.as_str(),
-        ) {
-            Some(event_abi) => event_abi,
-            None => {
-                return Box::new(future::err(format_err!(
-                    "Event with the signature \"{}\" not found in \
-                     contract \"{}\" of data source \"{}\"",
-                    event_handler.event,
-                    self.data_source_contract_abi.name,
-                    self.data_source_name
-                )));
-            }
-        };
+        let block = block.clone();
+        let transaction = transaction.clone();
+        let log = log.clone();
 
-        // Parse the log into an event
-        let params = match event_abi.parse_log(RawLog {
-            topics: log.topics.clone(),
-            data: log.data.clone().0,
-        }) {
-            Ok(log) => log.params,
+        let data_source_name = self.data_source_name.clone();
+        let abi_name = self.data_source_contract_abi.name.clone();
+        let contract = self.data_source_contract_abi.contract.clone();
+
+        // If there are no matching handlers, fail processing the event
+        let potential_handlers = match self.handlers_for_log(&log) {
+            Ok(handlers) => handlers,
             Err(e) => {
-                return Box::new(future::err(format_err!(
-                    "Failed to parse parameters of event: {}: {}",
-                    event_handler.event,
-                    e
-                )));
+                return Box::new(future::err(e)) as Box<dyn Future<Item = _, Error = _> + Send>
             }
         };
 
-        debug!(
-            logger, "Start processing Ethereum event";
-            "signature" => &event_handler.event,
-            "handler" => &event_handler.handler,
-            "data_source" => &self.data_source_name,
-            "address" => format!("{}", &log.address),
-        );
+        // Filter out handlers whose signatures don't exist in the contract ABI;
```

Comment:
"Filter out" sounds like this situation isn't an error, perhaps "Catch" or "Error on" would better.